homeassistant/components/hyundai_kia_connect/binary_sensor.py

```diff
@@ -0,0 +1,206 @@
+        key="back_left_door_is_open",
+        name="Back Left Door",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-door",
+        off_icon="mdi:car-door",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="back_right_door_is_open",
+        name="Back Right Door",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-door",
+        off_icon="mdi:car-door",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="trunk_is_open",
+        name="Trunk",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car-back",
+        off_icon="mdi:car-back",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="hood_is_open",
+        name="Hood",
+        device_class=BinarySensorDeviceClass.DOOR,
+        on_icon="mdi:car",
+        off_icon="mdi:car",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="ev_battery_is_charging",
+        name="EV Battery Charge",
+        device_class=BinarySensorDeviceClass.BATTERY_CHARGING,
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="ev_battery_is_plugged_in",
+        name="EV Battery Plug",
+        device_class=BinarySensorDeviceClass.PLUG,
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="fuel_level_is_low",
+        name="Fuel Low Level",
+        device_class=BinarySensorDeviceClass.LIGHT,
+        on_icon="mdi:gas-station-off",
+        off_icon="mdi:gas-station",
+    ),
+    HyundaiKiaBinarySensorEntityDescription(
+        key="data",
+        name="Debug Data",
+    ),
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up binary_sensor platform."""
+    coordinator = hass.data[DOMAIN][config_entry.unique_id]
+    entities = []
+    for vehicle_id in coordinator.data.vehicles.keys():
+        vehicle: Vehicle = coordinator.data.vehicles[vehicle_id]
+        for description in SENSOR_DESCRIPTIONS:
+            if getattr(vehicle, description.key, None) is not None:
+                entities.append(
+                    HyundaiKiaConnectBinarySensor(coordinator, description, vehicle)
+                )
+    async_add_entities(entities, True)
+    return True
+
+
+class HyundaiKiaConnectBinarySensor(BinarySensorEntity, HyundaiKiaConnectEntity):
+    """Hyundai / Kia Connect binary sensor class."""
+
+    def __init__(
+        self, coordinator, description: BinarySensorEntityDescription, vehicle: Vehicle
+    ):
+        """Initialize the sensor."""
+        HyundaiKiaConnectEntity.__init__(self, coordinator, vehicle)
+        self._description = description
+        self._key = self._description.key
+        self._attr_unique_id = f"{DOMAIN}_{vehicle.id}_{self._key}"
+        self._attr_name = f"{vehicle.name} {self._description.name}"
+        self._attr_device_class = self._description.device_class
+        if self._key == "data":
+            self._attr_extra_state_attributes = vehicle.data
```

Comment:
Why are we setting all data as state attributes?

I suggest we don't do this and, in case they are interesting, create entities instead.

```suggestion
```

Reply:
this is needed for debugging purposes, as there are mild differences in API responses between different cars, regions and brands. I would keep this on a disabled state and users can enable it if needed. Unless, this is something against `best practices` or `architectural principles`

Reply:
> this is needed for debugging purposes

Then we should use debug statements, and not pollute our state engine.

Reply:
debug statements are hard to receive for a newbie to home assistant and it includes to touch configuration.yaml to enable debugging. But, this sensor keeps the API response and it is always accessible. Our experience while working on multi-region, multi-car solution proved that having this ease the overall debugging easier. I propose to keep this as disabled by default and user can enable it whenever it is needed.

Moreover, some users notice that their cars supports different sensors and they can easily point out the sensor while sharing their data.

Reply:
anyway, to make progress, I am removing this.